cmd/instance_pool_create.go

```diff
@@ -0,0 +1,227 @@
+	ID              string   `json:"id"`
+	Name            string   `json:"description"`
+	Description     string   `json:"name"`
+	Serviceoffering string   `json:"serviceoffering"`
+	Template        string   `json:"template"`
+	Zone            string   `json:"zone"`
+	Affinitygroups  []string `json:"affinitygroups"`
+	Securitygroups  []string `json:"securitygroups"`
+	Privnets        []string `json:"Privnets"`
+	Keypair         string   `json:"keypair"`
+	Userdata        string   `json:"userdata"`
+	Size            int64    `json:"size"`
+	State           string   `json:"state"`
+}
+
+func (o *instancePoolCreateItemOutput) toJSON()  { outputJSON(o) }
+func (o *instancePoolCreateItemOutput) toText()  { outputText(o) }
+func (o *instancePoolCreateItemOutput) toTable() { outputTable(o) }
+
+var instancePoolCreateCmd = &cobra.Command{
+	Use:     "create <name>",
+	Short:   "Create an instance pool",
+	Aliases: gCreateAlias,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return cmd.Usage()
+		}
+
+		description, err := cmd.Flags().GetString("description")
+		if err != nil {
+			return err
+		}
+
+		zoneName, err := cmd.Flags().GetString("zone")
+		if err != nil {
+			return err
+		}
+
+		if zoneName == "" {
+			zoneName = gCurrentAccount.DefaultZone
+		}
+
+		zone, err := getZoneByName(zoneName)
+		if err != nil {
+			return err
+		}
+
+		so, err := cmd.Flags().GetString("service-offering")
+		if err != nil {
+			return err
+		}
+
+		servOffering, err := getServiceOfferingByName(so)
+		if err != nil {
+			return err
+		}
+
+		templateFilterCmd, err := cmd.Flags().GetString("template-filter")
+		if err != nil {
+			return err
+		}
+		templateFilter, err := validateTemplateFilter(templateFilterCmd)
+		if err != nil {
+			return err
+		}
+
+		templateName, err := cmd.Flags().GetString("template")
+		if err != nil {
+			return err
+		}
+
+		if templateName == "" {
+			templateName = gCurrentAccount.DefaultTemplate
+		}
+
+		template, err := getTemplateByName(zone.ID, templateName, templateFilter)
+		if err != nil {
+			return err
+		}
+
+		keypair, err := cmd.Flags().GetString("keypair")
+		if err != nil {
+			return err
+		}
+
+		if keypair == "" {
+			keypair = gCurrentAccount.DefaultSSHKey
+		}
+
+		size, err := cmd.Flags().GetInt("size")
+		if err != nil {
+			return err
+		}
+
+		sg, err := cmd.Flags().GetStringSlice("security-group")
+		if err != nil {
+			return err
+		}
+
+		sgs, err := getSecurityGroups(sg)
+		if err != nil {
+			return err
+		}
+
+		aff, err := cmd.Flags().GetStringSlice("anti-affinity-group")
+		if err != nil {
+			return err
+		}
+
+		affs, err := getAffinityGroup(aff)
+		if err != nil {
+			return err
+		}
+
+		priv, err := cmd.Flags().GetStringSlice("privnet")
+		if err != nil {
+			return err
+		}
+
+		privs, err := getPrivnetList(priv, zone.ID)
+		if err != nil {
+			return err
+		}
+
+		//It use asyncTasks to have spinner when user exec this command
```

Comment:
This comment is probably not necessary...

Reply:
fixed